### PR TITLE
feat: allow for host level etcd

### PR DESCRIPTION
**Pull request by andrewrynhard**
This allows users to create an etcd service using the host init system.

### CODE REVIEW

File: internal/app/machined/pkg/system/services/etcd.go [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+		}
+	}
+
+	// Set the process arguments.
+	args := runner.Args{
+		ID: e.ID(data),
+		ProcessArgs: []string{
+			"/usr/local/bin/etcd",
+			"--name=" + hostname,
+			"--listen-client-urls=https://0.0.0.0:2380",
```
Comment (Ulexus) [inline internal/app/machined/pkg/system/services/etcd.go]: We should probably make a package which takes an IP address (ips[0] in this case), checks whether it is IPv4 or IPv6, and offers AllAddress() (0.0.0.0 or ::) and LoopBackAddress() (127.0.0.1 or ::1) for each.

Comment (andrewrynhard) [reply internal/app/machined/pkg/system/services/etcd.go]: Good point 👍

File: internal/app/machined/pkg/system/services/kubelet.go [hunk 2]
```diff
@@ -53,6 +53,9 @@ func (k *Kubelet) Condition(data *userdata.UserData) conditions.Condition {
 
 // DependsOn implements the Service interface.
 func (k *Kubelet) DependsOn(data *userdata.UserData) []string {
+	if data.Services.Kubeadm.IsControlPlane() {
+		return []string{"containerd", "etcd", "kubeadm"}
```
Comment (andrewrynhard) [inline internal/app/machined/pkg/system/services/kubelet.go]: I wonder if we should depend on etcd here. Do we gain anything? Probably faster bootstrap times.

File: pkg/constants/constants.go [hunk 3]
```diff
@@ -115,17 +115,23 @@ const (
...
 	DefaultKubernetesVersion = "1.16.0"
 
 	// KubernetesImage is the enforced hyperkube image to use for the control plane.
 	KubernetesImage = "k8s.gcr.io/hyperkube"
 
+	// DefaultEtcdVersion is the default target version of etcd.
+	DefaultEtcdVersion = "3.3.15-0"
+
+	// EtcdImage is the reposistory for the etcd image.
+	EtcdImage = "k8s.gcr.io/etcd"
```
Comment (andrewrynhard) [inline pkg/constants/constants.go]: would love to use quay, but ran into an issue with the manifest version bing incompatible with containerd.

File: pkg/crypto/x509/x509.go [hunk 4]
```diff
@@ -78,6 +85,13 @@ type Options struct {
 // Option is the functional option func.
 type Option func(*Options)
 
+// CommonName sets the common name of the certificate.
+func CommonName(o string) Option {
```
Comment (andrewrynhard) [inline pkg/crypto/x509/x509.go]: Kubernetes PKI auth doesn't work without a CN.

File: internal/app/machined/pkg/system/services/kubelet.go (hunk 2 above)
Comment (andrewrynhard) [reply internal/app/machined/pkg/system/services/kubelet.go]: Nope. Kubeadm still depends on etcd. Maybe when we move to bootkube we will get this benefit.

**Comment by andrewrynhard** [comment]
Seeing the following in the etcd logs:

```
2019-10-03 03:05:40.578508 I | embed: rejected connection from "10.5.0.4:40210" (error "x509: certificate is valid for 10.5.0.2, 127.0.0.1, not 10.5.0.4", ServerName "", IPAddresses ["10.5.0.2" "127.0.0.1"], DNSNames [])
2019-10-03 03:05:42.936813 I | embed: rejected connection from "10.5.0.3:47532" (error "x509: certificate is valid for 10.5.0.2, 127.0.0.1, not 10.5.0.3", ServerName "", IPAddresses ["10.5.0.2" "127.0.0.1"], DNSNames [])
```

File: internal/app/machined/pkg/system/services/etcd.go (hunk 1 above)
Comment (andrewrynhard) [reply internal/app/machined/pkg/system/services/etcd.go]: @Ulexus gonna follow up with a IPv6/IPV4 dual stack helper library.

**Pull request merged**